### DISCUSSION TITLE
Error __dirname is not defined in edge runtime

### DISCUSSION THREAD

**Discussion by anc95** [vercel/community#1531] [Help] [answered]
I have a middleware which is pretty simple
```js
import { NextResponse } from 'next/server';

const config = {
  matcher: '/api/github/webhooks',
};

async function middleware(request) {
  const json = await request;
  if (!json) {
    return NextResponse.next();
  }
  if (
    json.action === 'opened' &&
    json.pull_request &&
    json.pull_request.state === 'open'
  ) {
    return NextResponse.next();
  }
  if (
    json.action === 'created' &&
    json.sender &&
    json.sender.type === 'User' &&
    json.comment &&
    json.comment.body &&
    json.comment.body.startsWith('/cr.gpt')
  ) {
    return NextResponse.next();
  }
  return NextResponse.redirect('https://github.com');
}

export { config, middleware };
```

Get the bellow error on the function logs.

<img width="864" alt="image" src="https://user-images.githubusercontent.com/13167934/218377765-cee9a490-c292-44e7-b116-72d6d321fd11.png">

I tried
1. compiled this to CJS
2. set `globalThis.__dirname = '/'`

But all didn't help me?

Anyone have any idea how I fix this, Thanks in advance

**Comment by anc95** [comment, marked as answer]
I solved this, the cause is that I was not using nextjs, so I should import next function from '@vercel/edge'

**Reply by ODelibalta** [reply]
can you please share your code? I am still fighting the issue

**Reply by anc95** [reply]
@ODelibalta https://github.com/anc95/ChatGPT-CodeReview/blob/main/middleware.ts

**Reply by ODelibalta** [reply]
@anc95 Thank you for this. `rewrite` is not working as well as the `redirect` but it will do for now. It is not actually appending parameters to the url and actually changing the url etc. But it seems to be working. I am getting a 404 this time but the `dirname` error is gone. Thank you
